File: components/skills/archimedes-omega/clients/java/src/main/java/org/arkhe/ArchimedesClient.java
package org.arkhe;

import okhttp3.*;
import com.google.gson.Gson;
import java.io.IOException;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ArchimedesClient {
    private final OkHttpClient httpClient;
    private final String baseUrl;
    private final Gson gson = new Gson();

    public ArchimedesClient(String baseUrl) {
        this.baseUrl = baseUrl;
        this.httpClient = new OkHttpClient();
    }

    public String analyze(Object request) throws IOException {
        String json = gson.toJson(request);
        RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
        Request httpRequest = new Request.Builder()
                .url(baseUrl + "/analyze")
                .post(body)
                .build();
        try (Response response = httpClient.newCall(httpRequest).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String simulateSU2(Object request) throws IOException {
        String json = gson.toJson(request);
        RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
        Request httpRequest = new Request.Builder()
                .url(baseUrl + "/simulate/su2")
                .post(body)
                .build();
        try (Response response = httpClient.newCall(httpRequest).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String simulateSL3Z(Object request) throws IOException {
        String json = gson.toJson(request);
        RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
        Request httpRequest = new Request.Builder()
                .url(baseUrl + "/simulate/sl3z")
                .post(body)
                .build();
        try (Response response = httpClient.newCall(httpRequest).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String simulateWState(Object request) throws IOException {
        String json = gson.toJson(request);
        RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
        Request httpRequest = new Request.Builder()
                .url(baseUrl + "/simulate/wstate")
                .post(body)
                .build();
        try (Response response = httpClient.newCall(httpRequest).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String detectPeaks(Object request) throws IOException {
        String json = gson.toJson(request);
        RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
        Request httpRequest = new Request.Builder()
                .url(baseUrl + "/detect/peaks")
                .post(body)
                .build();
        try (Response response = httpClient.newCall(httpRequest).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String checkTeleportationResource(List<Double> phases, List<Double> coherence, int nodes, double lossProb) throws IOException {
        Map<String, Object> payload = new HashMap<>();
        payload.put("phases", phases);
        payload.put("coherence", coherence);
        payload.put("nodes", nodes);
        payload.put("loss_probability", lossProb);

        String json = gson.toJson(payload);
        RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
        Request httpRequest = new Request.Builder()
                .url(baseUrl + "/analyze/teleportation-resource")
                .post(body)
                .build();
        try (Response response = httpClient.newCall(httpRequest).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }
}

File: components/skills/archimedes-omega/clients/dotnet/ArchimedesClient.cs
using System;
using System.Threading.Tasks;
using RestSharp;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace Arkhe.ArchimedesAgent
{
    public class ArchimedesClient
    {
        private readonly RestClient _client;
        public ArchimedesClient(string baseUrl) => _client = new RestClient(baseUrl);

        public async Task<string> AnalyzeAsync(object request)
        {
            var req = new RestRequest("/analyze", Method.Post);
            req.AddJsonBody(request);
            var response = await _client.ExecuteAsync(req);
            return response.Content;
        }

        public async Task<string> SimulateSU2Async(object request)
        {
            var req = new RestRequest("/simulate/su2", Method.Post);
            req.AddJsonBody(request);
            var response = await _client.ExecuteAsync(req);
            return response.Content;
        }

        public async Task<string> SimulateSL3ZAsync(object request)
        {
            var req = new RestRequest("/simulate/sl3z", Method.Post);
            req.AddJsonBody(request);
            var response = await _client.ExecuteAsync(req);
            return response.Content;
        }

        public async Task<string> SimulateWStateAsync(object request)
        {
            var req = new RestRequest("/simulate/wstate", Method.Post);
            req.AddJsonBody(request);
            var response = await _client.ExecuteAsync(req);
            return response.Content;
        }

        public async Task<string> DetectPeaksAsync(object request)
        {
            var req = new RestRequest("/detect/peaks", Method.Post);
            req.AddJsonBody(request);
            var response = await _client.ExecuteAsync(req);
            return response.Content;
        }

        public async Task<string> CheckTeleportationResourceAsync(List<double> phases, List<double> coherence, int nodes = 3, double lossProb = 0.2)
        {
            var req = new RestRequest("/analyze/teleportation-resource", Method.Post);
            req.AddJsonBody(new {
                phases = phases,
                coherence = coherence,
                nodes = nodes,
                loss_probability = lossProb
            });
            var response = await _client.ExecuteAsync(req);
            return response.Content;
        }
    }
}
